### DISCUSSION TITLE
Binding helper #forEach question

### DISCUSSION THREAD

**Discussion by tglynx** [Budibase/budibase#11561] [Help] [answered]
Hi,

i want to display a list of 'ClubMemberships' from a budibase relation in a paragraph. 
I have a dataprovider that returns a single row. 
I can access the ClubMembership relation like this using a binding: 
`{{ DataProviderProfile.Rows.0.ClubMemberships }}`

It returns something like this: 
`[{"_id":"ro_ta_21b0e9d84615462f884a64e030c30037_b0e3e92c19ed4f69a9e49f541a93853b","primaryDisplay":"Membership 1"},{"_id":"ro_ta_21b0e9d84615462f884a64e030c30037_d13a1362d0d544858abcd9a9c0fd4302","primaryDisplay":"Membership 2"}]`

According to this helper documentation
<img width="708" alt="Bildschirmfoto 2023-08-19 um 12 55 58" src="https://github.com/Budibase/budibase/assets/30650435/21e3598c-2409-4e2b-b418-030feaec946b">
i would assume the following to work:
`{{ #forEach DataProviderProfile.Rows.0.ClubMemberships }} {{ primaryDisplay }} {{ /forEach }}` 

But it is not... ultimately i would like to join the items using ', ' exept it is the last one using isLast somehow (according to the documentation).

How is this done?

**Comment by tglynx** [comment, marked as answer]
After some searching i found the solution to be: 
`{{#each DataProviderProfile.Rows.0.ClubMemberships }}{{ primaryDisplay }} {{#unless @last}}, {{/unless}} {{/each}}`
Found here: https://handlebarsjs.com/guide/builtin-helpers.html#each

Even better:
`{{#if DataProviderProfile.Rows.0.ClubMemberships }}
  {{#each DataProviderProfile.Rows.0.ClubMemberships }}{{ primaryDisplay }} {{#unless @last}}, {{/unless}} {{/each}}
{{else}}
  No Memberships!
{{/if}}`

**Reply by melohagan** [reply]
Personally I find JavaScript a bit easier for loops and if statements. In this case:

```js
return `$("DataProviderProfile.Rows")[0]["ClubMemberships"]?.map(membership => membership.primaryDisplay) || "No Memberships!"
```
